added purge dashboard test

sunilangadi2 opened the pull request:
Signed-off-by: Sunil Angadi <sangadi@redhat.com>

# Description
In RHCEPH 4.3 onwards to purge ceph dashboard we have new playbook called purge-dashboard.yml
hence added purge dashboard test in existing suite.
bz: [1786691](https://bugzilla.redhat.com/show_bug.cgi?id=1786691)
polarion link:[ CEPH-83573277](https://polarion.engineering.redhat.com/polarion/#/project/CEPH/workitem?id=CEPH-83573277)
Success logs: [cephci-run-WFK8QW](http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-WFK8QW/)

Please include Automation development guidelines. Source of Test case - New Feature/Regression Test/Close loop of customer BZs

# Checklist:

- [ ] Create a test case in Polarion reviewed and approved.
- [ ] Create a design/automation approach doc. Optional for tests with similar tests already automated.
- [ ] Review the automation design
- [ ] Implement the test script and perform test runs
- [ ] Submit PR for code review and approve
- [ ] Update Polarin Test with Automation script details and update automation fields
- [ ] If automation is part of Close loop, update BZ flag qe-test_coverage “+” and link Polarion test

sunilkumarn417 commented on tests/ceph_ansible/purge_dashboard.py:
```diff
@@ -0,0 +1,56 @@
+
+
+def run(**kw):
+    """
```
please follow google docstring style.
https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html

sunilkumarn417 commented on tests/ceph_ansible/purge_dashboard.py:
```diff
@@ -0,0 +1,56 @@
+    """
+
+    Args:
+       param kw:
```
please explain `kw` in example section.

sunilkumarn417 commented on tests/ceph_ansible/purge_dashboard.py:
```diff
@@ -0,0 +1,56 @@
+log = logging.getLogger(__name__)
+
+
+def run(**kw):
```
please check other test modules, Hope you may need ceph_cluster to point which exact cluster in case multiple ceph clusters.

sunilangadi2 replied:
ceph cluster nodes we can get it from **kw itself, https://github.com/red-hat-storage/cephci/blob/31922863cb5cf994cd27a6427174d734156cf81c/tests/ceph_ansible/purge_cluster.py#L28
So i don't think we need ceph_cluster also as param.

sunilkumarn417 commented on tests/ceph_ansible/purge_dashboard.py:
```diff
@@ -0,0 +1,56 @@
+""" Module to purge ceph dashboard"""
```
please update module level doc string to end `.`

sunilkumarn417 commented on tests/ceph_ansible/purge_dashboard.py:
```diff
@@ -0,0 +1,56 @@
+    log.info("Running purge-dashboard test")
+    ceph_nodes = kw.get("ceph_nodes")
+    installer_node = None
+    ansible_dir = "/usr/share/ceph-ansible"
```
As mentioned in the doc string `ansible-dir`, value is not considered here.
I think it would be 
```
ansible_dir = kw.get("ansible-dir", "/usr/share/ceph-ansible")
```

sunilangadi2 replied:
Since these path is very static, So i thought it's better to hard code in module itself,
Instead of adding everytime in needed test suites 
@sunilkumarn417 Are you ok with this?

sunilkumarn417 replied:
Oh okay, review comment was based on `ansible-dir` key-value in `kw` doc string.
If it is not changing anytime, can we take out that parameter from doc string.

sunilkumarn417 commented on tests/ceph_ansible/purge_dashboard.py:
```diff
@@ -0,0 +1,56 @@
+    installer_node = None
+    ansible_dir = "/usr/share/ceph-ansible"
+    inventory = "hosts"
+    playbook_command = (
```
same comment on `playbook-command` param.

sunilkumarn417 commented on tests/ceph_ansible/purge_dashboard.py:
```diff
@@ -0,0 +1,56 @@
+
+    # executing the command to purge dashboard
+    out, err = installer_node.exec_command(cmd=cmd, long_running=True)
+    log.info("STDOUT:\n %s" % out)
```
can we make use of f-strings?

sunilkumarn417 replied:
exec_command generate log records incrementally because of `long_running=True`. do we need to log STDOUT again?

sunilkumarn417 commented on tests/ceph_ansible/purge_dashboard.py:
```diff
@@ -0,0 +1,56 @@
+    cmd = f"cd {ansible_dir} ; ansible-playbook -vvvv -i {inventory} {playbook} {playbook_options}"
+
+    # executing the command to purge dashboard
+    out, err = installer_node.exec_command(cmd=cmd, long_running=True)
```
can we rename `err` to `rc`, Since it is exit status.

Ameenasuhani commented on suites/nautilus/dashboard/sanity_containerized_ceph_dashboard.yaml:
```diff
@@ -84,6 +84,12 @@ tests:
             timeout: 300
       desc: Perform rgw tests
 
+   - test:
+      name: Purge ceph dashboard
+      module: purge_dashboard.py
+      polarion-id: CEPH-83573277
+      desc: Purges ceph dashboard
+
```
@sunilangadi2 We can use already existing purge something like this instead of creating a new  purge_dashboard.py https://github.com/red-hat-storage/cephci/blob/master/suites/nautilus/ansible/sanity_containerized_ceph_ansible.yaml#L185-L192

https://github.com/red-hat-storage/cephci/blob/master/tests/ceph_ansible/purge_cluster.py#L33-L39

sunilangadi2 replied:
@Ameenasuhani Yes, know this `purge_cluster.py` to support purge cluster playbook but here lot of other things like 
https://github.com/red-hat-storage/cephci/blob/31922863cb5cf994cd27a6427174d734156cf81c/tests/ceph_ansible/purge_cluster.py#L52-L58
and 
https://github.com/red-hat-storage/cephci/blob/31922863cb5cf994cd27a6427174d734156cf81c/tests/ceph_ansible/purge_cluster.py#L100-L105
there in order to handle purge-cluster playbook which are not needed for purge-dashboard playbook, hence created new `purge_dashboard` module.
So hope you are Ok with this, Please let me know your thoughts?

Ameenasuhani replied:
@sunilangadi2 the code you are pointing to is just removing certain packages post purge. what I'm I missing here?

sunilangadi2 replied:
@Ameenasuhani that part of code is not needed for dashboard-purge playbook, otherwise it would be messy in that module

sunilkumarn417 reviewed (commented):
Hi @sunilangadi2 ,

PR looks good.
But there are few minor comments and if possible can we add validation to ensure ceph-dashboard is removed from cluster and as well as from ceph configuration.

-Thanks

sunilangadi2 commented:
@sunilkumarn417 Thanks for the review, addressed all comments also added validations steps.
Please review.
Success logs: http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-TLWZKW/

sunilkumarn417 commented on tests/ceph_ansible/purge_dashboard.py:
```diff
@@ -0,0 +1,63 @@
+    """To purge ceph dashboard module.
+
+    Args:
+        **kw: Arbitrary keyword arguments.
```
please explain about `kw` in example section.

sunilkumarn417 commented on tests/ceph_ansible/purge_dashboard.py:
```diff
@@ -0,0 +1,63 @@
+
+
+def run(ceph_cluster, **kw):
+    """To purge ceph dashboard module.
```
I think it should be `purge ceph dashboard module.`

sunilkumarn417 reviewed (commented):
Hi @sunilangadi2 

Thanks for addressing the review comments.
I have some minor comments on doc-string.

Sorry, I have missed this out in the first review.

sunilangadi2 commented:
@sunilkumarn417 addresed doc review please review it.
